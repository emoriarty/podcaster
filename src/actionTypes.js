export const FETCH_COUNTRIES = 'itunes/FETCH_COUNTRIES';
export const FETCH_COUNTRIES_SUCCESS = 'itunes/FETCH_COUNTRIES_SUCCESS';
export const FETCH_COUNTRIES_FAIL = 'itunes/FETCH_COUNTRIES_FAIL';

export const FETCH_MEDIA_TYPES = 'itunes/FETCH_MEDIA_TYPES';
export const FETCH_MEDIA_TYPES_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_SUCCESS';
export const FETCH_MEDIA_TYPES_FAIL = 'itunes/FETCH_MEDIA_TYPES_FAIL';

export const FETCH_COMMON_TRANSLATIONS = 'itunes/FETCH_COMMON_TRANSLATIONS';
export const FETCH_COMMON_TRANSLATIONS_SUCCESS = 'itunes/FETCH_COMMON_TRANSLATIONS_SUCCESS';
export const FETCH_COMMON_TRANSLATIONS_FAIL = 'itunes/FETCH_COMMON_TRANSLATIONS_FAIL';

export const FETCH_MEDIA_TYPES_TRANSLATIONS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_SUCCESS';
export const FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL = 'itunes/FETCH_MEDIA_TYPES_TRANSLATIONS_FAIL';
