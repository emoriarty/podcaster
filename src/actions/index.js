export { default as fetchCountries } from './itunes/fetchCountries';
export { default as fetchMediaTypes } from './itunes/fetchMediaTypes';
