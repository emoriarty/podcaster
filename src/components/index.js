export * from './Expansion'
export * from './Layouts'
export * from './ListItems'
export * from './Texts'
