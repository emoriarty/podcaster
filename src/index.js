import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import configStore from './store'
import injectTapEventPlugin from 'react-tap-event-plugin'
import App from './containers/App'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import '../styles/main.scss'

injectTapEventPlugin()

const store = configStore()

ReactDOM.render(
  <Provider store={store}>
    <MuiThemeProvider>
      <App />
    </MuiThemeProvider>
  </Provider>,
  document.getElementById('app')
)
