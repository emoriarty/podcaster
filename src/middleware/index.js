export { default as yqlMiddleware } from './yql'
export { default as rssMiddleware } from './rss'
