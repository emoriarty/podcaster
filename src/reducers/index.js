import { combineReducers } from 'redux'
import countries from './countries'
import entities from './entities'
import mediaTypes from './mediaTypes'
import settings from './settings'
import translations from './translations'
import podcasts from './podcasts'

const app = combineReducers({
  entities,
  countries,
  mediaTypes,
  settings,
  translations,
  podcasts
})

export default app
