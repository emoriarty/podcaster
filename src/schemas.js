import { schema } from 'normalizr'

export const country = new schema.Entity(
  'countries',
  {},
  { idAttribute: 'country_code' }
)

export const mediaType = new schema.Entity('mediaTypes')

export const podcast = new schema.Entity(
  'podcasts',
  {},
  {
    idAttribute: (value, parent, key) => {
      return value.id.attributes['im:id']
    }
  })

export const podcastFeed = { feed: { entry: new schema.Array(podcast) } }
