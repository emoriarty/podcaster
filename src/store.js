import { createStore, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'
import { yqlMiddleware } from './middleware'
import appReducer from './reducers'

const loggerMiddleware = createLogger()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

export default () =>
  createStore(
    appReducer,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        yqlMiddleware,
        loggerMiddleware
      )
    )
  )
