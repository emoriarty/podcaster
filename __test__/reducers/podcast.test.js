import reducer from '../../src/reducers/podcasts'
import * as types from '../../src/actionTypes'
import jsonfile from 'jsonfile'
import path from 'path'

const mockResponse = jsonfile.readFileSync(path.resolve(__dirname, '../../__mocks__/podcasts.json'))

describe('podcasts reducer', () => {
  it('should return the initial state', () =>
    expect(reducer(undefined, {}))
      .toMatchSnapshot()
  )

  it('when action FETCH_PODCASTS sets isFecthing property to true', () => {
    const action = {
      type: types.FETCH_PODCASTS
    }
    const next = reducer(undefined, action)
    expect(next.isFetching)
      .toBeTruthy()
  })

  it('should handle FETCH_PODCASTS_SUCCESS', () => {
    const action = {
      type: types.FETCH_PODCASTS_SUCCESS,
      payload: { result: mockResponse }
    }
    expect(reducer([], action))
      .toMatchSnapshot()
  })

  it('should handle FETCH_PODCASTS_FAIL', () => {
    const action = {
      type: types.FETCH_PODCASTS_FAIL,
      error: new Error()
    }
    expect(reducer([], action))
      .toMatchSnapshot()
  })
})
