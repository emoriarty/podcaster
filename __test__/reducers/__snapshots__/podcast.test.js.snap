// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`podcasts reducer should handle FETCH_PODCASTS_FAIL 1`] = `
Object {
  "error": [Error],
  "isFetching": false,
}
`;

exports[`podcasts reducer should handle FETCH_PODCASTS_SUCCESS 1`] = `
Object {
  "ids": Array [
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1309",
          "label": "Cine y TV",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-cine-y-tv/id1309?mt=2&uo=2",
          "term": "TV & Film",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "1220451382",
        },
        "label": "https://itunes.apple.com/es/podcast/bienvenido-a-la-vida-peligrosa/id1220451382?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/podium-podcast/id1118080197?mt=2&uo=2",
        },
        "label": "Podium podcast",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is2.mzstatic.com/image/thumb/Podcasts122/v4/66/9d/95/669d95b7-549e-f703-77e9-d8f4b7228b41/mza_7824967408217628104.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is3.mzstatic.com/image/thumb/Podcasts122/v4/66/9d/95/669d95b7-549e-f703-77e9-d8f4b7228b41/mza_7824967408217628104.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is2.mzstatic.com/image/thumb/Podcasts122/v4/66/9d/95/669d95b7-549e-f703-77e9-d8f4b7228b41/mza_7824967408217628104.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Bienvenido a la vida peligrosa",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "02/06/2017",
        },
        "label": "2017-06-02T02:20:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/bienvenido-a-la-vida-peligrosa/id1220451382?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "Una serie sonora con guión original de Arturo Pérez-Reverte y dirigida por Guillermo Arriaga, donde se mezclan la filosofía, el sexo, la violencia y el narcotráfico. Una historia de la que es imposible escapar.",
      },
      "title": Object {
        "label": "Bienvenido a la vida peligrosa - Podium podcast",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1303",
          "label": "Comedia",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-comedia/id1303?mt=2&uo=2",
          "term": "Comedy",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "665143133",
        },
        "label": "https://itunes.apple.com/es/podcast/nadie-sabe-nada/id665143133?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-ser/id131600381?mt=2&uo=2",
        },
        "label": "Cadena SER (CadenaSer)",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts111/v4/9e/85/9e/9e859e5d-a54b-3981-fe73-a652e1cae30b/mza_8058364541956303330.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts111/v4/9e/85/9e/9e859e5d-a54b-3981-fe73-a652e1cae30b/mza_8058364541956303330.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts111/v4/9e/85/9e/9e859e5d-a54b-3981-fe73-a652e1cae30b/mza_8058364541956303330.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Nadie sabe nada",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "02/06/2017",
        },
        "label": "2017-06-02T08:38:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/nadie-sabe-nada/id665143133?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "Improvisación en estado puro, con Andreu Buenafuente y Berto Romero",
      },
      "title": Object {
        "label": "Nadie sabe nada - Cadena SER (CadenaSer)",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1303",
          "label": "Comedia",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-comedia/id1303?mt=2&uo=2",
          "term": "Comedy",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "976066920",
        },
        "label": "https://itunes.apple.com/es/podcast/oh-my-lol/id976066920?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-ser/id131600381?mt=2&uo=2",
        },
        "label": "Cadena SER",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts111/v4/8c/d6/91/8cd6919e-4376-7b73-2565-db5764a7f228/mza_987268868711370480.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts111/v4/8c/d6/91/8cd6919e-4376-7b73-2565-db5764a7f228/mza_987268868711370480.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts111/v4/8c/d6/91/8cd6919e-4376-7b73-2565-db5764a7f228/mza_987268868711370480.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Oh My LOL",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "05/06/2017",
        },
        "label": "2017-06-05T06:43:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/oh-my-lol/id976066920?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "El programa de humor que cambiará las noches de la radio",
      },
      "title": Object {
        "label": "Oh My LOL - Cadena SER",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1316",
          "label": "Deportes y ocio",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-deportes-y-ocio/id1316?mt=2&uo=2",
          "term": "Sports & Recreation",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "1146392357",
        },
        "label": "https://itunes.apple.com/es/podcast/el-partidazo-de-cope/id1146392357?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-cope/id212045712?mt=2&uo=2",
        },
        "label": "Cadena COPE",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is4.mzstatic.com/image/thumb/Podcasts111/v4/fd/15/49/fd154980-d133-385b-e749-16e9b8620205/mza_5542246499824712450.png/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts111/v4/fd/15/49/fd154980-d133-385b-e749-16e9b8620205/mza_5542246499824712450.png/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts111/v4/fd/15/49/fd154980-d133-385b-e749-16e9b8620205/mza_5542246499824712450.png/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "El Partidazo de COPE",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "05/06/2017",
        },
        "label": "2017-06-05T14:59:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/el-partidazo-de-cope/id1146392357?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "De lunes a viernes de 23:30h a 01:30h con Juanma Castaño",
      },
      "title": Object {
        "label": "El Partidazo de COPE - Cadena COPE",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1316",
          "label": "Deportes y ocio",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-deportes-y-ocio/id1316?mt=2&uo=2",
          "term": "Sports & Recreation",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "389954489",
        },
        "label": "https://itunes.apple.com/es/podcast/tiempo-de-juego/id389954489?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-cope/id212045712?mt=2&uo=2",
        },
        "label": "Cadena COPE",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts127/v4/5c/40/eb/5c40eb50-9a3b-ede5-a388-851ad2c0c8f9/mza_3759756348697878797.png/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts127/v4/5c/40/eb/5c40eb50-9a3b-ede5-a388-851ad2c0c8f9/mza_3759756348697878797.png/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is4.mzstatic.com/image/thumb/Podcasts127/v4/5c/40/eb/5c40eb50-9a3b-ede5-a388-851ad2c0c8f9/mza_3759756348697878797.png/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Tiempo de Juego",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "04/06/2017",
        },
        "label": "2017-06-04T14:59:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/tiempo-de-juego/id389954489?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "Sábado y domingo de 15:00h a 01:30h con Paco González, Manolo Lama y Pepe Domingo Castaño",
      },
      "title": Object {
        "label": "Tiempo de Juego - Cadena COPE",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1324",
          "label": "Cultura y sociedad",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-cultura-y-sociedad/id1324?mt=2&uo=2",
          "term": "Society & Culture",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "468637823",
        },
        "label": "https://itunes.apple.com/es/podcast/la-rosa-de-los-vientos/id468637823?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/onda-cero/id350577649?mt=2&uo=2",
        },
        "label": "OndaCero",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is3.mzstatic.com/image/thumb/Podcasts71/v4/44/a7/ff/44a7fff0-7be0-da46-b561-be64f8d2c914/mza_2203792290179665825.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is4.mzstatic.com/image/thumb/Podcasts71/v4/44/a7/ff/44a7fff0-7be0-da46-b561-be64f8d2c914/mza_2203792290179665825.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts71/v4/44/a7/ff/44a7fff0-7be0-da46-b561-be64f8d2c914/mza_2203792290179665825.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "La rosa de los vientos",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "04/06/2017",
        },
        "label": "2017-06-04T23:45:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/la-rosa-de-los-vientos/id468637823?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "rights": Object {
        "label": "Copyright © Atresmedia Corporación de Medios de Comunicación, S.A - www.antena3.com",
      },
      "summary": Object {
        "label": "Podcast de Historia, misterio, Investigación, relatos. Sábados 1:00 a 4:00 Domingos 1:30 a 04:00 con Bruno Cardeñosa, Fernando Rueda, Martin Exposito, Jesús Callejo.",
      },
      "title": Object {
        "label": "La rosa de los vientos - OndaCero",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1303",
          "label": "Comedia",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-comedia/id1303?mt=2&uo=2",
          "term": "Comedy",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "902539655",
        },
        "label": "https://itunes.apple.com/es/podcast/la-vida-moderna/id902539655?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-ser/id131600381?mt=2&uo=2",
        },
        "label": "Cadena SER",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts111/v4/6e/a5/e9/6ea5e90c-bc47-6123-1a92-5c9adb47e0fe/mza_2479348688310110700.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts111/v4/6e/a5/e9/6ea5e90c-bc47-6123-1a92-5c9adb47e0fe/mza_2479348688310110700.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is3.mzstatic.com/image/thumb/Podcasts111/v4/6e/a5/e9/6ea5e90c-bc47-6123-1a92-5c9adb47e0fe/mza_2479348688310110700.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "La vida moderna",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "28/12/2015",
        },
        "label": "2015-12-28T22:43:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/la-vida-moderna/id902539655?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "David Broncano viaja por España para investigar una variedad de asuntos desde un cierto desconocimiento pero siempre con torería",
      },
      "title": Object {
        "label": "La vida moderna - Cadena SER",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1303",
          "label": "Comedia",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-comedia/id1303?mt=2&uo=2",
          "term": "Comedy",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "1122106243",
        },
        "label": "https://itunes.apple.com/es/podcast/educa-como-puedas/id1122106243?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/podium-podcast/id1118080197?mt=2&uo=2",
        },
        "label": "Podium podcast",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is3.mzstatic.com/image/thumb/Podcasts71/v4/b7/68/0d/b7680d7a-2039-74f4-0160-62be856d11f4/mza_9189825939190537947.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is4.mzstatic.com/image/thumb/Podcasts71/v4/b7/68/0d/b7680d7a-2039-74f4-0160-62be856d11f4/mza_9189825939190537947.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts71/v4/b7/68/0d/b7680d7a-2039-74f4-0160-62be856d11f4/mza_9189825939190537947.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Educa Como Puedas",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "28/07/2016",
        },
        "label": "2016-07-28T02:10:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/educa-como-puedas/id1122106243?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "Ser padre o madre supone pasar a ser patrimonio público de la Humanidad. Es decir, todos te van a criticar y nunca vas a hacer nada bien. Pero con Carles Capdevilla, al menos, te vas a reír. Un podcast de monólogos para Podium.",
      },
      "title": Object {
        "label": "Educa Como Puedas - Podium podcast",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1469",
          "label": "Cursos de idiomas",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-educaci%C3%B3n-cursos-de-idiomas/id1469?mt=2&uo=2",
          "term": "Language Courses",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "262026989",
        },
        "label": "https://itunes.apple.com/es/podcast/the-english-we-speak/id262026989?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/bbc/id121676617?mt=2&uo=2",
        },
        "label": "BBC Radio",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is5.mzstatic.com/image/thumb/Podcasts71/v4/67/9f/b2/679fb2e8-31ac-f532-1229-ede5dd6b437c/mza_3784815412610190712.jpg/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts71/v4/67/9f/b2/679fb2e8-31ac-f532-1229-ede5dd6b437c/mza_3784815412610190712.jpg/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is1.mzstatic.com/image/thumb/Podcasts71/v4/67/9f/b2/679fb2e8-31ac-f532-1229-ede5dd6b437c/mza_3784815412610190712.jpg/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "The English We Speak",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "29/05/2017",
        },
        "label": "2017-05-29T17:00:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/the-english-we-speak/id262026989?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "rights": Object {
        "label": "© (C) BBC 2017",
      },
      "summary": Object {
        "label": "Every week, we look at a different everyday English phrase or piece of slang in this fun three-minute programme. From BBC World Service",
      },
      "title": Object {
        "label": "The English We Speak - BBC Radio",
      },
    },
    Object {
      "category": Object {
        "attributes": Object {
          "im:id": "1311",
          "label": "Noticias y política",
          "scheme": "https://itunes.apple.com/es/genre/podcasts-noticias-y-pol%C3%ADtica/id1311?mt=2&uo=2",
          "term": "News & Politics",
        },
      },
      "id": Object {
        "attributes": Object {
          "im:id": "1035694177",
        },
        "label": "https://itunes.apple.com/es/podcast/herrera-en-cope/id1035694177?mt=2&uo=2",
      },
      "im:artist": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/artist/cadena-cope/id212045712?mt=2&uo=2",
        },
        "label": "Cadena COPE",
      },
      "im:contentType": Object {
        "attributes": Object {
          "label": "Podcast",
          "term": "Podcast",
        },
      },
      "im:image": Array [
        Object {
          "attributes": Object {
            "height": "55",
          },
          "label": "http://is2.mzstatic.com/image/thumb/Podcasts122/v4/08/ee/04/08ee04cd-a519-712b-e4f6-f12203fed40e/mza_4784014713839064973.png/55x55bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "60",
          },
          "label": "http://is3.mzstatic.com/image/thumb/Podcasts122/v4/08/ee/04/08ee04cd-a519-712b-e4f6-f12203fed40e/mza_4784014713839064973.png/60x60bb-85.jpg",
        },
        Object {
          "attributes": Object {
            "height": "170",
          },
          "label": "http://is2.mzstatic.com/image/thumb/Podcasts122/v4/08/ee/04/08ee04cd-a519-712b-e4f6-f12203fed40e/mza_4784014713839064973.png/170x170bb-85.jpg",
        },
      ],
      "im:name": Object {
        "label": "Herrera en COPE",
      },
      "im:price": Object {
        "attributes": Object {
          "amount": "0",
          "currency": "EUR",
        },
        "label": "Obtener",
      },
      "im:releaseDate": Object {
        "attributes": Object {
          "label": "06/06/2017",
        },
        "label": "2017-06-06T00:29:00-07:00",
      },
      "link": Object {
        "attributes": Object {
          "href": "https://itunes.apple.com/es/podcast/herrera-en-cope/id1035694177?mt=2&uo=2",
          "rel": "alternate",
          "type": "text/html",
        },
      },
      "summary": Object {
        "label": "De lunes a viernes de 06:00h a 13:00h con Carlos Herrera",
      },
      "title": Object {
        "label": "Herrera en COPE - Cadena COPE",
      },
    },
  ],
  "isFetching": false,
}
`;

exports[`podcasts reducer should return the initial state 1`] = `
Object {
  "error": null,
  "ids": Array [],
  "isFetching": false,
}
`;
